python/tests/simplify.py

```diff
@@ -303,6 +321,51 @@ def merge_labeled_ancestors(self, S, input_id):
             if num_edges == 0 and not is_sample:
                 self.rewind_node(input_id, output_id)
 
+    def extract_ancestry(self, edge):
+        S = []
+        x = self.A_head[edge.child]
+
+        x_head = None
+        x_prev = None
+        while x is not None:
+            if x.right > edge.left and edge.right > x.left:
```

Comment:
What happens if this condition is false ie. if there is *no* overlap between `edge` and `x`? In this case presumably you would want `x` to remain unchanged in `A`, but it looks as if you remove it regardless of whether this `if` condition is true or not (lines 365-366 further down). I couldn't see a test that addressed this fringe case explicitly but I'll have another look now

Reply:
Oh actually nevermind, I realise that's the case your lines 357 - 3461 are covering here